builder/src/main/java/cz/xtf/builder/db/PostgreSQL.java

```diff
@@ -1,47 +1,138 @@
 
-    public PostgreSQL(boolean withLivenessProbe, boolean withReadinessProbe, boolean withStartupProbe) {
-        super("POSTGRESQL", "/var/lib/pgsql/data", withLivenessProbe, withReadinessProbe, withStartupProbe, true);
-    }
+    // env variables names for the Red Hat image
+    private static final String DEFAULT_POSTGRESQL_USER_ENV_VAR = "POSTGRESQL_USER";
+    private static final String DEFAULT_POSTGRESQL_DATABASE_ENV_VAR = "POSTGRESQL_DATABASE";
 
-    public PostgreSQL(PersistentVolumeClaim pvc) {
-        super("POSTGRESQL", "/var/lib/pgsql/data", pvc);
-    }
+    // env variables names for the Official Docker PostgreSQL image
+    private static final String OFFICIAL_IMAGE_POSTGRESQL_USER_ENV_VAR = "POSTGRES_USER";
+    private static final String OFFICIAL_IMAGE_POSTGRESQL_DATABASE_ENV_VAR = "POSTGRES_DB";
+    private static final String OFFICIAL_IMAGE_POSTGRES_PASSWORD_ENV_VAR = "POSTGRES_PASSWORD";
+
+    // default env variables for the Red Hat image
+    private static final Map<String, String> DEFAULT_VARS = new HashMap<String, String>() {
+        {
+            put("POSTGRESQL_MAX_CONNECTIONS", "100");
+            put("POSTGRESQL_SHARED_BUFFERS", "16MB");
+            put("POSTGRESQL_MAX_PREPARED_TRANSACTIONS", "90");
+            // Temporary workaround for https://github.com/sclorg/postgresql-container/issues/297
+            // Increase the "set_passwords.sh" timeout from the default 60s to 300s to give the
+            // PostgreSQL server chance properly to start under high OCP cluster load
+            put("PGCTLTIMEOUT", "300");
+        }
+    };
 
-    public PostgreSQL(PersistentVolumeClaim pvc, boolean withLivenessProbe, boolean withReadinessProbe) {
-        super("POSTGRESQL", "/var/lib/pgsql/data", pvc, withLivenessProbe, withReadinessProbe);
+    // default env variables for for the Official Docker PostgreSQL image
+    private static final Map<String, String> OFFICIAL_IMAGE_DEFAULT_VARS = new HashMap<String, String>() {
+        {
+            // Temporary workaround for https://github.com/sclorg/postgresql-container/issues/297
+            // Increase the "set_passwords.sh" timeout from the default 60s to 300s to give the
+            // PostgreSQL server chance properly to start under high OCP cluster load
+            put("PGCTLTIMEOUT", "300");
+        }
+    };
+
+    // default command arguments for the Official Docker PostgreSQL image
+    private static final List<String> OFFICIAL_IMAGE_DEFAULT_ARGS = new ArrayList<String>() {
+        {
+            add("-c");
+            add("shared_buffers=16MB");
+            add("-c");
+            add("max_connections=100");
+            add("-c");
+            add("max_prepared_transactions=90");
+        }
+    };
+
+    private String postgresqlUserEnvVar;
+    private String postgresqlDatabaseEnvVar;
+    private Map<String, String> vars;
+    private List<String> args;
+    private boolean isOfficialImage;
```

Comment:
I assume whole complexity of this PR comes from attempt to support 2 kind of postgres images.

What I would expect @mnovak1 would want is whatever changes we make are backward compatible. So if someone is using redhat images do not need to change anything. @mnovak1 am I right?

Those changes here in this PR does not look like that kind of changes. At least because of Removal of existing constructors.

Reply:
> Those changes here in this PR does not look like that kind of changes. At least because of Removal of existing constructors.

Right, we need to keep backward compatibility. API must not change.

Reply:
I suspected that.... I'll revert deleting the constructors .... even though we should probably get rid of them eventually because the current code is a mess expecially when you try and figure out what the default values are....

Reply:
reverted

Reply:
Constructors returned back. Resolving this thread.